app/javascript/liquidTags/userSubscriptionLiquidTag.js

```diff
@@ -0,0 +1,310 @@
+    if (isVisible) {
+      instructions.classList.remove('hidden');
+    } else {
+      instructions.classList.add('hidden');
+    }
+  }
+}
+
+function submitSubscription() {
+  toggleSubmitButtonsState({ disabled: true, textContent: 'Submitting' });
+
+  const articleId =
+    document.getElementById('article-body')?.dataset?.articleId ?? null;
+
+  const subscriber = userData();
+  const body = JSON.stringify({
+    user_subscription: {
+      source_type: 'Article',
+      source_id: articleId,
+      subscriber_email: subscriber.email,
+    },
+  });
+
+  return getCsrfToken()
+    .then(sendFetch('user_subscriptions', body))
+    .then((res) => res.json());
+}
+
+function fetchSubscriptionStatus() {
+  const { articleId } = document.getElementById('article-body').dataset;
+
+  const params = new URLSearchParams({
+    source_type: 'Article',
+    source_id: articleId,
+  }).toString();
+
+  const headers = {
+    Accept: 'application/json',
+    'X-CSRF-Token': window.csrfToken,
+    'Content-Type': 'application/json',
+  };
+
+  return fetch(`/user_subscriptions/subscribed?${params}`, {
+    method: 'GET',
+    headers,
+    credentials: 'same-origin',
+  }).then((response) => {
+    if (response.ok) {
+      return response.json();
+    }
+    console.error(
+      `Base data error: ${response.status} - ${response.statusText}`,
+    );
+  });
+}
+
+function showSubscribedNotices() {
+  const { authorUsername } = document.getElementsByClassName(
+    'ltag__user-subscription-tag',
+  )[0].dataset;
+
+  updateNotices({
+    variant: 'success',
+    content: `You are now subscribed and may receive emails from ${authorUsername}`,
+  });
+}
+
+function populateSubscriberProfileImage({
+  profile_image_90: profileImageSrc,
+  username,
+}) {
+  document
+    .querySelectorAll('.ltag__user-subscription-tag__subscriber-profile-image')
+    .forEach((profileImage) => {
+      profileImage.src = profileImageSrc;
+      profileImage.alt = username;
+    });
+}
+
+/**
+ * Initializes the functionality of any user_subscription liquid tags in an article.
+ *
+ * An article may have more than one user_subscription liquid tag, but all instances complete the same action,
+ * and are therefore kept in-step throughout any user interaction or update (i.e. the UI and behavior of all user_subscription
+ * liquid tags in an article will always be the same).
+ */
+export async function initializeUserSubscriptionLiquidTagContent() {
+  const allUserSubLiquidTags = document.getElementsByClassName(
+    'ltag__user-subscription-tag__container',
+  );
+
+  const { userStatus } = document.querySelector('body').dataset;
+  if (userStatus === 'logged-out') {
+    for (const liquidTag of allUserSubLiquidTags) {
+      initSignedOutState(liquidTag);
+    }
+    return;
+  }
+
+  const { currentUser: user } = await getUserDataAndCsrfToken();
+  const { email } = user;
+  const isSubscriberAuthedWithApple =
+    email && email.endsWith('@privaterelay.appleid.com');
```

Comment:
I've copied the logic from the original liquid tag script, which shows a prompt to change your email in settings if you  have authed with apple... but we don't actually return a user's `email` in the `base_data` request, so this condition is never going to be met.

Instead, what is happening for a user with this format of email is they see the 'subscribe' button, but get this error when they try to subscribe:
![Subscriber email Can't subscribe with Apple private relay. Please update email.](https://user-images.githubusercontent.com/20773163/152985516-9884e2c0-1f96-483d-98e0-61ad8373269b.png)

Paging @fdocr / @joshpuetz - is there another way to detect if the logged in user is authed with apple?

Reply:
Right now I can't think of another way to identify this (on the frontend) other than using the email. In the end we don't need to know if they have SIWA enabled, what we really need to know is whether or not we'll be able to send them emails (they could have SIWA enabled but kept their email).

> but we don't actually return a user's email in the base_data request, so this condition is never going to be met.

I may have lost some context overtime here, but the `getUserDataAndCsrfToken()` utility only parses the JSON stored in `body.dataset.user`. This user data gets populated in the first place by the `/async_info/base_data` endpoint, which [renders this data here](https://github.com/forem/forem/blob/aitchiss/subscribe-btn-issues/app/controllers/async_info_controller.rb#L39-L63).

Is there a reason we don't want to include the email here? My first guess would be privacy concerns around having that on the DOM (easy to scrape by extensions maybe), but if the user navigates to their settings or profile (if email is public) then it would also appear there. Any other reasoning here?

Two possible solutions for this that come to mind (both adding something to the [payload here]()):

1. Add the user email to the payload
   - This code would remain unchanged but the email would now be there (this may have been the case in the original implementation)
2. Add a boolean value to the payload to identify if they can be reached via email
   - The check for `endsWith('@privaterelay.appleid.com')` would move to the backend
   - In the payload we already have booleans like this one (i.e. [admin](https://github.com/forem/forem/blob/aitchiss/subscribe-btn-issues/app/controllers/async_info_controller.rb#L60))

I think option 2 would be slightly better, but any thoughts/suggestions/preferences here?

Reply:
TBH, tracing back the pull requests for this feature, I'm not sure the appleAuth message part of it ever worked 😄  It looks like a few years ago there was a TODO pending some changes to email confirmation that never got fixed, and the original test steps for the feature required locally adding the user email to the base_data 🤷‍♀️ 

Anyway... I think option 2 sounds pretty good. It doesn't feel like there's a need to send the user's email address, and like you say there are similar booleans coming through that payload too.

Thanks for the advice - much appreciated!